src/cache/cache_las.c

```diff
@@ -1073,20 +1073,23 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
             cnt = 0;
 
         WT_ERR(__wt_compare(session, NULL, &las_key, &cache->las_max_key, &cmp));
-        if (las_id > cache->las_max_btree_id || (las_id == cache->las_max_btree_id && cmp > 0) ||
-          (las_id == cache->las_max_btree_id && cmp == 0 && las_counter > cache->las_max_counter)) {
+        if (las_btree_id > cache->las_max_btree_id ||
+          (las_btree_id == cache->las_max_btree_id && cmp > 0) ||
+          (las_btree_id == cache->las_max_btree_id && cmp == 0 &&
+              las_timestamp > cache->las_max_timestamp) ||
+          (las_btree_id == cache->las_max_btree_id && cmp == 0 &&
+              las_timestamp == cache->las_max_timestamp && las_txnid > cache->las_max_txnid)) {
             __wt_buf_free(session, sweep_key);
             ret = WT_NOTFOUND;
             break;
         }
 
         /*
-         * We only want to break between key blocks. Stop if we've processed enough entries either
-         * all we wanted or enough and there is a reader waiting and we're on a key boundary.
+         * Stop if we've processed enough entries either all we wanted or enough and there is a
+         * reader waiting.
          */
         ++visit_cnt;
-        if (!removing_key_block &&
```

Comment:
@sulabhM I've just removed this part of the check. To my understanding, it's just there to avoid breaking out midway through removing a block and having half of an old block in the las.

Reply:
Can't just remove this. We're going to be dealing with sweep properly later.